wgpu-hal/src/gles/adapter.rs

```diff
@@ -293,6 +321,22 @@ impl super::Adapter {
             ver >= (3, 1),
         );
 
+        // Check platforms effected by mesa bug:
+        // https://gitlab.freedesktop.org/mesa/mesa/-/issues/2565.
+        //
+        // This comment
+        // (https://gitlab.freedesktop.org/mesa/mesa/-/merge_requests/4972/diffs?diff_id=75888#22f5d1004713c9bbf857988c7efb81631ab88f99_323_327)
+        // seems to indicate all skylake models are effected, so we check for renderer models that
+        // have their 3 letter code ending with `L`
+        let r = renderer.to_lowercase();
+        if r.contains("mesa")
+            && r.contains("intel")
+            && r.split(&[' ', '(', ')'][..])
+                .any(|substr| substr.len() == 3 && substr.chars().nth(2) == Some('l'))
+        {
+            private_caps.set(super::PrivateCapabilities::REQUIRES_SHADER_CLEAR, true);
```

Comment:
We should also issue a log::warn when we enable this. Something like:

"Detected skylake derivative running on mesa i915. Clears to srgb framebuffers will use manual shader clears."